cores/esp32/HardwareSerial.cpp

```diff
@@ -367,6 +367,7 @@ void HardwareSerial::begin(unsigned long baud, uint32_t config, int8_t rxPin, in
 #endif
             default:
                 log_e("Bad UART Number");
+                HSERIAL_MUTEX_UNLOCK();
```

Comment:
This default case is here just to make the compiler happy. In fact, you will never get here with wrong port number, because this check has already been made on line 330

Reply:
I just tested the compilation and it doesn't complain about the lack of `default:` case.
I'll just remove this case from the code to make it clearer.